packages/toc/src/tocitem.tsx

```diff
@@ -44,7 +45,7 @@ export class TableOfContentsItem extends React.PureComponent<
     const { children, isActive, heading, onCollapse, onMouseDown } = this.props;
 
     return (
-      <li className="jp-tocItem">
+      <TreeItem className="jp-tocItem">
         <div
           className={`jp-tocItem-heading ${
             isActive ? 'jp-tocItem-active' : ''
```

Comment:
`isActive` prop needs to be use not to set a class but to set the tree item `selected` attribute.
`heading.collapsed` will be needed to set the `expanded` attribute.
You can remove the `button` with the collapser icon.

`<TreeItem selected={isActive} expanded={!heading.collapsed} ...>